ADAL/src/cache/ADTokenCacheItem.m

```diff
@@ -48,55 +47,71 @@ - (void)calculateHash
 //we create a special, "broad" cache item, with nil resource and access token:
 - (BOOL)isMultiResourceRefreshToken
 {
-    return [NSString adIsStringNilOrBlank:self.resource]
-        && [NSString adIsStringNilOrBlank:self.accessToken]
-       && ![NSString adIsStringNilOrBlank:self.refreshToken];
+    return [NSString adIsStringNilOrBlank:_resource]
+        && [NSString adIsStringNilOrBlank:_accessToken]
+       && ![NSString adIsStringNilOrBlank:_refreshToken];
 }
 
 - (id)copyWithZone:(NSZone*) zone
 {
-    ADTokenCacheItem* item = [[self.class allocWithZone:zone] init];
+    ADTokenCacheItem* item = [[ADTokenCacheItem allocWithZone:zone] init];
     
-    item.resource = [self.resource copyWithZone:zone];
-    item.authority = [self.authority copyWithZone:zone];
-    item.clientId = [self.clientId copyWithZone:zone];
-    item.familyId = [self.familyId copyWithZone:zone];
-    item.accessToken = [self.accessToken copyWithZone:zone];
-    item.accessTokenType = [self.accessTokenType copyWithZone:zone];
-    item.refreshToken = [self.refreshToken copyWithZone:zone];
-    item.expiresOn = [self.expiresOn copyWithZone:zone];
-    item.userInformation = [self.userInformation copyWithZone:zone];
-    item.sessionKey = [self.sessionKey copyWithZone:zone];
-    item.tombstone = self.tombstone ? [NSMutableDictionary dictionaryWithDictionary:self.tombstone] : nil;
+    item->_resource = [_resource copyWithZone:zone];
+    item->_authority = [_authority copyWithZone:zone];
+    item->_clientId = [_clientId copyWithZone:zone];
+	item->_familyId = [_familyId copyWithZone:zone];
+    item->_accessToken = [_accessToken copyWithZone:zone];
+    item->_accessTokenType = [_accessTokenType copyWithZone:zone];
+    item->_refreshToken = [_refreshToken copyWithZone:zone];
+    item->_expiresOn = [_expiresOn copyWithZone:zone];
+    item->_userInformation = [_userInformation copyWithZone:zone];
+    item->_sessionKey = [_sessionKey copyWithZone:zone];
+	item->_tombstone = [_tombstone mutableCopyWithZone:zone];
```

Comment:
if _tombstone is nil, we want the copy value also be nil, so should it be 
item->_tombstone = _tombstone ? [_tombstone mutableCopyWithZone:zone] : nil;
?